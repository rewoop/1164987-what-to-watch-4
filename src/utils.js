import {ALL_GENRES, RatingLevel} from "./const.js";

export const formatReviewDate = (dateString) => {
  const date = new Date(dateString);
  const dateTimeFormat = new Intl.DateTimeFormat(`en`, {year: `numeric`, month: `2-digit`, day: `2-digit`});
  const [{value: month},, {value: day},, {value: year}] = dateTimeFormat.formatToParts(date);
  return `${year}-${month}-${day}`;
};

export const extend = (a, b) => {
  return Object.assign({}, a, b);
};

export const getGenresList = (films) => {
  return [ALL_GENRES].concat(Array.from(new Set(films.map((film) => film.filmGenre))));
};

export const formatRatingScoreToLevel = (score) => {
  if (score >= RatingLevel.MIN_BAD && score <= RatingLevel.MAX_BAD) {
    return `Bad`;
  } else if (score > RatingLevel.MAX_BAD && score <= RatingLevel.MAX_NORMAL) {
    return `Normal`;
  } else if (score > RatingLevel.MAX_NORMAL && score <= RatingLevel.MAX_GOOD) {
    return `Good`;
  } else if (score > RatingLevel.MAX_GOOD && score < RatingLevel.MAX_VERY_GOOD) {
    return `Very good`;
  } else {
    return `Awesome`;
  }
};
