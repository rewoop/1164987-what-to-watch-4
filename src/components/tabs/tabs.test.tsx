import * as React from "react";
import * as renderer from "react-test-renderer";
import Tabs from "./tabs";
import {noop} from "../../utils";

const Settings = {
  currentTab: `overview`
};

it(`Should Tabs render correctly`, () => {
  const tree = renderer
    .create(<Tabs
      onLinkClickHandler={noop}
      currentTab={Settings.currentTab}
    />).toJSON();

  expect(tree).toMatchSnapshot();
});
