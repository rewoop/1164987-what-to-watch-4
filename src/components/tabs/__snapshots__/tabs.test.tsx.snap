// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Tabs render correctly 1`] = `
<ul
  className="movie-nav__list"
>
  <li
    className="movie-nav__item movie-nav__item--active"
    onClick={[Function]}
  >
    <a
      className="movie-nav__link"
      style={
        Object {
          "cursor": "pointer",
        }
      }
    >
      Overview
    </a>
  </li>
  <li
    className="movie-nav__item"
    onClick={[Function]}
  >
    <a
      className="movie-nav__link"
      style={
        Object {
          "cursor": "pointer",
        }
      }
    >
      Details
    </a>
  </li>
  <li
    className="movie-nav__item"
    onClick={[Function]}
  >
    <a
      className="movie-nav__link"
      style={
        Object {
          "cursor": "pointer",
        }
      }
    >
      Reviews
    </a>
  </li>
</ul>
`;
